### DISCUSSION TITLE
Access token problems with a private repository in organization

### DISCUSSION THREAD

**Discussion by DaanV2** [community/community#24378] [Apps, API and Webhooks]
<p>For the organization, we have an in house desktop application that sometimes has the download the contents of a GitHub repository. each user has generated a user token and given it to the application which stores the token.</p>
<p>But when the application tries to download a private repository from the organization they are all met with a 404. this indiciates by github FAQ that they dont have access. as github doesn’t tell if the repo exists or not. But it does work for my access token?</p>
<p>So I was wondering how to get it to work for everyone</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hey <a class="mention" href="/daanv2">@DaanV2</a> o/</p>
<p>How are you going? Curious to know if this is still a concern, as I imagine it might be.</p>
<p>Since you’ve developed an app, I’m curious to know the rationale around using PATs fed to the app, as opposed to providing auth for the app, like:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/developers/apps/authenticating-with-github-apps" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/developers/apps/authenticating-with-github-apps" target="_blank" rel="noopener">Authenticating with GitHub Apps - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Considering Personal Access Tokens are tied to specific users, there is some concerning consideration to make with how those PATs are handled by your app. Are they being stored in a database and are they encrypted?</p>
<p>Regarding your 404, your understanding is correct. As long as the auth isn’t bumping up against any other issues (like syntax) we would assume that a PAT returning a 404, does not have permissions to make that particular action.</p>
<p>So for your specific ask:</p>
<blockquote>
<p>So I was wondering how to get it to work for everyone</p>
</blockquote>
<p>I’d be curious to know the justification on pulling PATs instead of having a unique auth token for the app itself. There may be a use case I’m not considering!</p>
